src/v/cluster/archival_metadata_stm.cc

```diff
@@ -650,6 +786,8 @@ ss::future<std::error_code> archival_metadata_stm::do_replicate_commands(
         co_return result.error();
     }
 
+    _last_replicated_offset = batch_last_offset;
```

Comment:
I don't understand this, the offsets are assigned after the replication happens so in general case `batch.last_offset()` is undefined. Why don't you set `_last_replicated_offset` to `result.value().last_offset` after the replication happened?

Reply:
Good spot - thank you. Fixed.